### PR TITLE
fix: cURL not detecting the right body type

### PR DESCRIPTION
**Pull request by sondermanish**
## Description

This commit fixes the bug https://github.com/appsmithorg/appsmith/issues/13978 and changes two files:
- CurlImporterServiceTest.java
- CurlImporterServiceCEImpl.java

Additionally this commit fixes the Body type detection for REST APIs with GET method  when imported from cURL.

Fixes https://github.com/appsmithorg/appsmith/issues/13978

## Type of change

- Bug fix (Now the cURL import will be able to detect the right body type)

## How Has This Been Tested?
- This PR has been tested:
        * Manually
        * Junit
        
- As the Junit coverage is adequate, we don't require Cypress test case for this PR. 

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/CurlImporterServiceCEImpl.java [hunk 1]
```diff
@@ -50,6 +51,8 @@ public class CurlImporterServiceCEImpl extends BaseApiImporter implements CurlIm
     private static final String ARG_COOKIE = "--cookie";
     private static final String ARG_USER = "--user";
     private static final String ARG_USER_AGENT = "--user-agent";
+    private static final String FORM_DATA_KEY = "apiContentType";
+    private static final String CONTENT_TYPE = "content-type";
```
Comment (nidhi-nair) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/CurlImporterServiceCEImpl.java:55]: @manish535 I get that we're switching to lower case here for consistent behaviour with what the client understands, but can we make the client more lenient with how it understands the header as well? HTTP headers are meant to be [case insensitive](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers) and we should not force this rigidity at any layer.

Also, in general it is recommended to retain the casing that was originally present here for readability and standardization, so if anything we should only be making client side changes for this fix.

File: app/server/appsmith-server/src/test/java/com/appsmith/server/services/CurlImporterServiceTest.java [hunk 2]
```diff
@@ -335,7 +336,7 @@ public void multilineCommand() throws AppsmithException {
 
         final ActionConfiguration actionConfiguration = action.getActionConfiguration();
         assertEmptyPath(action);
-        assertHeaders(action, new Property("Content-Type", "application/json"));
```
Comment (nidhi-nair) [inline app/server/appsmith-server/src/test/java/com/appsmith/server/services/CurlImporterServiceTest.java:338]: Also, were you able to figure out why the client is using the header and not the `formdata.apiContentType` property to render the correct body tab?

Comment (sondermanish) [reply app/server/appsmith-server/src/test/java/com/appsmith/server/services/CurlImporterServiceTest.java]: The front-end doesn't use this property as of now to make detect the right body type, rather it is dependent on the parsed header for right detection, the formData.apiContentType is used for detecting the Body type when there is get request, if it is null then GET request will be sent without Body, verified with manual testing. Have made changes to address that.

File: app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/CurlImporterServiceCEImpl.java (hunk 1 above)
Comment (sondermanish) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/CurlImporterServiceCEImpl.java]: @nidhi-nair Have made changes accordingly, This PR is not complete as the issue will persist as long as Front-end doesn't add  leniency in header checks. Assigning the issue to @AmanAgarwal041.

**Comment by nidhi-nair** [comment]
Closing in favour of #16549

**Pull request closed without merging**